src/Sylius/Bundle/CoreBundle/Twig/CustomerCancelOrderExtension.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Twig;
+
+use Sylius\Bundle\CoreBundle\Checker\CustomerOrderCancellationCheckerInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+
+final class CustomerCancelOrderExtension extends \Twig_Extension implements CustomerCancelOrderExtensionInterface
```

Comment:
Maybe instead of this class, we check [state machine guards](https://github.com/winzou/state-machine/blob/master/examples/simple.php#L39-L44) and use regular `sm_can` helper